x/collection/genesis.go

```diff
@@ -0,0 +1,154 @@
+package collection
+
+import (
+	codectypes "github.com/line/lbm-sdk/codec/types"
+
+	sdk "github.com/line/lbm-sdk/types"
+	sdkerrors "github.com/line/lbm-sdk/types/errors"
+
+	"github.com/line/lbm-sdk/x/token/class"
+)
+
+const (
+	DefaultDepthLimit = 3
+	DefaultWidthLimit = 8
+)
+
+// ValidateGenesis check the given genesis state has no integrity issues
+func ValidateGenesis(data GenesisState) error {
+
+	// the legacy module did not validate the data.
+	// if LegacyMode {
+	// 	return nil
+	// }
+
+	for _, contract := range data.Contracts {
+		if err := class.ValidateID(contract.ContractId); err != nil {
+			return err
+		}
+
+		if err := validateName(contract.Name); err != nil {
+			return err
+		}
+		if err := validateBaseImgURI(contract.BaseImgUri); err != nil {
+			return err
+		}
+		if err := validateMeta(contract.Meta); err != nil {
+			return err
+		}
+	}
+
+	for _, contractClasses := range data.Classes {
+		if err := class.ValidateID(contractClasses.ContractId); err != nil {
+			return err
+		}
+
+		if len(contractClasses.Classes) == 0 {
+			return sdkerrors.ErrInvalidRequest.Wrap("classes cannot be empty")
+		}
+		for i := range contractClasses.Classes {
+			any := &contractClasses.Classes[i]
+			class := TokenClassFromAny(any)
```

Comment:
AFAIK, golang does not have a keyword `class`.

Reply:
Ah, I confused. There is `github.com/line/lbm-sdk/x/token/class` package in this file, so it can collides.

Reply:
There would be no problem because local variables take precedence. And, I removed `import class` in the latest commit, so there shouldn't be any aesthetic issues either.